### PR TITLE
Gap 2267: unsubscribe clean up

### PR DESCRIPTION
**Pull request by ConorFayleTCO**
## Description

Ticket # and link

Summary of the changes and the related issue. List any dependencies that are required for this change:

## Type of change

Please check the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

**Review by john-tco — APPROVED** [review]

**Comment by john-tco** [comment]
Approved - but we are going to need a similar unsubscribe reference cleanup fn within saved_search and grant_subscriptions

**Review by dylanwrightCO — APPROVED** [review]

File: src/service/newsletter/newsletter-subscription-service.ts [hunk 1]
```diff
@@ -66,12 +66,17 @@ export class NewsletterSubscriptionService {
...
-  async unsubscribeFromNewsletter(
-    plaintextEmail: string,
-    type: NewsletterType,
-    unsubscribeReferenceId?: string,
-    sub?: string,
-  ): Promise<void> {
+  async unsubscribeFromNewsletter({
+    plaintextEmail,
+    type,
+    unsubscribeReferenceId,
```
Comment (john-tco) [inline src/service/newsletter/newsletter-subscription-service.ts:72]: If a user unsubscribes via the front-end this value is not going to be populated. We need some logic in here to get the unsubscribe reference by the user id and newsletter type and use that to delete the reference.

**Review by john-tco — DISMISSED** [review]
We need to remove the unsubscribe reference row using the userId & notification id rather than the reference (so we can unsubscribe via the front-end and through the "magic-link" email. This will prevent the unsubscribe reference table from indefinitely growing in size. 

We also need to do this for grant subscriptions and saved searches (that part might not be this ticket?)

File: src/service/newsletter/newsletter-subscription-service.ts (hunk 1 above)
Comment (ConorFayleTCO) [reply src/service/newsletter/newsletter-subscription-service.ts:72]: aye I've got that in here: https://github.com/cabinetoffice/gap-find-backend/pull/69

**Review by dylanwrightCO — APPROVED** [review]

**Pull request merged**